crocket/src/main/java/com/crocket/Surface.java

```diff
@@ -0,0 +1,18 @@
+package com.crocket;
+
+public enum Surface {
+    ICE("crocket/assets/textures/IceTile.png"),
+    SAND("crocket/assets/textures/SandTile.png"),
+    GRASS("crocket/assets/textures/GrassTile.png");
+
+    private final String texturePath;
+
+    private Surface(String texturePath) {
```

Comment:
What is this constructor used for?

Reply:
We use the constructor to associate a texture path with an enum. While we cannot use the new operator to pass arguments to the constructor of enums. Java has a special syntax called declaration list.  So when we create the the enums we associate the texture-path attribute to the texture-path from the declaration list. This is done so that LevelView can use getFieldTexturePath(). 

https://www.baeldung.com/java-enum-values

Reply:
I do not understand either but it works